biz.aQute.bndlib/src/aQute/bnd/build/WorkspaceExternalPluginHandler.java

```diff
@@ -196,6 +204,121 @@ public void abort() {
 	}
 
 	@Override
-	public void close() {}
+	public void close() {
+		loaders.values()
+			.forEach(IO::close);
+	}
+
+	/**
+	 * Returns list of external plugin proxies that implement the given
+	 * interface. The proxies will load the actual plugin on demand when used.
+	 * That is, the plugins will be quite cheap unless used.
+	 *
+	 * @param interf the interface listed in `-plugin`.
+	 * @param attrs the attributes from the that interface, the name specifies
+	 *            the name of the plugin, wildcards allowed
+	 * @return a list of plugins loaded from the external plugin set
+	 */
+	@SuppressWarnings("unchecked")
+	public Result<List<Object>> getImplementations(Class<?> interf, Attrs attrs) {
+		assert interf.isInterface();
+
+		try {
+
+			String filter = ExternalPluginNamespace.filter(attrs.getOrDefault("name", "*"), interf);
+			List<Capability> externalCapabilities = workspace
+				.findProviders(ExternalPluginNamespace.EXTERNAL_PLUGIN_NAMESPACE, filter)
+				.collect(Collectors.toList());
+
+			List<Object> plugins = new ArrayList<>();
+			for (Capability c : externalCapabilities) {
+				Memoize<Object> delegate = Memoize.supplier(() -> load(c, attrs).unwrap());
+				Object proxy = Proxy.newProxyInstance(interf.getClassLoader(), new Class[] {
+					interf, AutoCloseable.class
+				}, new InvocationHandler() {
+
+					@Override
+					public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+
+						if (method.getDeclaringClass() == Object.class) {
+							return method.invoke(this, args);
+						}
+
+						if (method.getDeclaringClass() == AutoCloseable.class) {
```

Comment:
Not sure how useful this is? I think there are many possible changes where this will backfire since the new default method expects a value to be returned? I've changed it to invoke the method instead of calling close but this will fail if the delegate is not yet there or does not implement AutoCloseable